packages/odyssey-react-mui/src/theme/useContrastMode.test.tsx

```diff
@@ -0,0 +1,254 @@
+    });
+
+    afterEach(() => {
+      window.getComputedStyle = originalGetComputedStyle;
+    });
+
+    it("returns the background color of the element if it is not transparent", () => {
+      const element = document.createElement("div");
+      element.style.backgroundColor = "rgb(255, 0, 0)";
+      const result = getBackgroundColor(element);
+      expect(result).toBe("rgb(255, 0, 0)");
+    });
+
+    it("returns the background color of the parent if the element is transparent", () => {
+      const parent = document.createElement("div");
+      const child = document.createElement("div");
+      parent.appendChild(child);
+      parent.style.backgroundColor = "rgb(0, 255, 0)";
+
+      // Update mock for this specific test
+      window.getComputedStyle = jest
+        .fn()
+        .mockImplementation((el: HTMLElement) => ({
+          backgroundColor:
+            el === parent ? "rgb(0, 255, 0)" : "rgba(0, 0, 0, 0)",
+        }));
+
+      expect(getBackgroundColor(child)).toBe("rgb(0, 255, 0)");
+    });
+
+    it('returns "#ffffff" if no non-transparent background is found', () => {
+      const element = document.createElement("div");
+      expect(getBackgroundColor(element)).toBe("#ffffff");
+    });
+
+    it("returns HueNeutral50 token for its RGB equivalent", () => {
+      const element = document.createElement("div");
+      element.style.backgroundColor = hueNeutral50Rgb;
+
+      window.getComputedStyle = jest.fn().mockImplementation(() => ({
+        backgroundColor: hueNeutral50Rgb,
+      }));
+
+      expect(getBackgroundColor(element)).toBe(Tokens.HueNeutral50);
+    });
+
+    it("handles nested transparent elements correctly", () => {
+      const grandparent = document.createElement("div");
+      const parent = document.createElement("div");
+      const child = document.createElement("div");
+      grandparent.appendChild(parent);
+      parent.appendChild(child);
+      grandparent.style.backgroundColor = "rgb(0, 0, 255)";
+
+      window.getComputedStyle = jest
+        .fn()
+        .mockImplementation((el: HTMLElement) => ({
+          backgroundColor:
+            el === grandparent ? "rgb(0, 0, 255)" : "rgba(0, 0, 0, 0)",
+        }));
+
+      expect(getBackgroundColor(child)).toBe("rgb(0, 0, 255)");
+    });
+  });
+
+  describe("MutationObserver functionality", () => {
+    let originalAddEventListener: typeof document.addEventListener;
+    let originalRemoveEventListener: typeof document.removeEventListener;
+
+    beforeEach(() => {
+      originalAddEventListener = document.addEventListener;
+      originalRemoveEventListener = document.removeEventListener;
+      document.addEventListener = jest.fn();
+      document.removeEventListener = jest.fn();
+    });
+
+    afterEach(() => {
+      document.addEventListener = originalAddEventListener;
+      document.removeEventListener = originalRemoveEventListener;
+    });
+
+    it("should clean up observers and event listeners on unmount", () => {
```

Comment:
This can be part of the main `useContrastMode` hook tests because it's part of that hook.

What I'd like to see are the `beforeEach` and `afterEach` moved into the hook using the `jest.spyOn` syntax.